### PR TITLE
pythonPackages.scikit-bio: init at 0.5.4

### PR DESCRIPTION
**Pull request by costrouc**
###### Things done

pythonPackages.CacheControl: init at 0.12.5 

pythonPackages.hdmedians: init at 0.13 

pythonPackages.lockfile: init at 0.12.2 

pythonPackages.scikit-bio: init at 0.5.4 

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/lockfile/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ pkgs
```
Comment (FRidh) [inline pkgs/development/python-modules/lockfile/default.nix]: typically we don't pass in `pkgs`, only `lib`.

File: pkgs/development/python-modules/hdmedians/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+buildPythonPackage rec {
+  version = "0.13";
+  pname = "hdmedians";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "230f80e064d905c49a1941af1b7e806e2f22b3c9a90ad5c21fd17d72636ea277";
+  };
+
+  buildInputs = [ nose cython ];
```
Comment (FRidh) [inline pkgs/development/python-modules/hdmedians/default.nix]: nose is a test runner and should go into `checkInputs`

File: pkgs/development/python-modules/lockfile/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+buildPythonPackage rec {
+  pname = "lockfile";
+  version = "0.12.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "6aed02de03cba24efabcd600b30540140634fc06cfa603822d508d5361e9f799";
+  };
+
+  buildInputs = [ pbr nose ];
```
Comment (FRidh) [inline pkgs/development/python-modules/lockfile/default.nix]: nose

File: pkgs/development/python-modules/scikit-bio/default.nix [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    inherit pname version;
+    sha256 = "3243f1995ef24643c09ff4d9391a79528aadd8232e5aa5d66c38d7b2e0c92f24";
+  };
+
+  buildInputs = [ cython ];
+  checkInputs = [ coverage ];
+  propagatedBuildInputs = [ lockfile CacheControl decorator ipython matplotlib natsort numpy pandas scipy hdmedians scikitlearn ];
+
+  # remove on when version > 0.5.4
+  patchPhase = ''
```
Comment (FRidh) [inline pkgs/development/python-modules/scikit-bio/default.nix]: postPatch

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/hdmedians/default.nix (hunk 2 above)
Comment (costrouc) [reply pkgs/development/python-modules/hdmedians/default.nix]: fixed.

File: pkgs/development/python-modules/lockfile/default.nix (hunk 1 above)
Comment (costrouc) [reply pkgs/development/python-modules/lockfile/default.nix]: fixed

File: pkgs/development/python-modules/lockfile/default.nix (hunk 3 above)
Comment (costrouc) [reply pkgs/development/python-modules/lockfile/default.nix]: fixed

File: pkgs/development/python-modules/scikit-bio/default.nix (hunk 4 above)
Comment (costrouc) [reply pkgs/development/python-modules/scikit-bio/default.nix]: fixed.

**Pull request merged**